src/System.Runtime/tests/System/Reflection/SignatureTypes.netcoreapp.cs

```diff
@@ -491,7 +500,16 @@ private static void TestSignatureTypeInvariants(Type type)
 
                 int position = type.GenericParameterPosition;
                 Assert.True(position >= 0);
-                Assert.Equal("!!" + position, type.Name);
+
+                if (type.IsGenericTypeParameter)
+                {
+                    throw new Exception("Unexpected: There is no mechanism at this time to create Signature Types of generic parameters on types.");    
```

Comment:
Why an explicit exception rather than Assert.False?

Reply:
No special reason - we don't have `Assert.Fail` and the choice between two slightly forced alternatives to it is pretty arbitrary in my mind.

Reply:
Why do we need Assert.Fail here?  I don't understand why this isn't just:
```C#
Assert.False(type.IsGenericTypeParameter);
Assert.True(type.IsGenericMethodParameter);
Assert.Equal("!!" + position, type.Name);
```

Reply:
In every place other than signature types, generic parameters come in two flavors and it looks weird to just assert that it's only one of the two without comment especially when it's mingled among four or five other assertions. So I made it clearer by calling out that special case and where we'd put the new asserts if we ever did support the creation of signture type generic parameters.